### PR TITLE
[NS-2]: feat/POST endpoint to start upload

### PR DESCRIPTION
**Pull request by fabioespinosa**
Validates `uploads` array (coming from body), gets the download link for each from farmer and saves them in mongo new `upload` collection.

Check https://github.com/internxt/service-storage-models/pull/1 on storage models as well.

### CODE REVIEW

File: lib/server/routes/buckets.js [hunk 1]
```diff
@@ -1396,6 +1399,109 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+
+    const response = [];
+
+    async.eachLimit(
+      uploads,
+      1,
+      (upload, cb) => {
+        const { index, size } = upload;
+
+        if (typeof size !== 'number' || size < 0) {
```
Comment (sg-gs) [inline lib/server/routes/buckets.js]: Move all validations to the beginning (using a for or something like that) in order to:
- Avoid inserting an upload before ensuring all the entire data is ok. If the 3rd upload is not okay, you will have two uploads stalled
- Having the validations located in one point, thus, at the time of starting to use layers, will be easier to decouple validations from business logic and / or persistence

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: lib/server/routes/buckets.js (hunk 1 above)
Comment (fabioespinosa) [reply lib/server/routes/buckets.js]: Agree, on it 👍

Comment (fabioespinosa) [reply lib/server/routes/buckets.js]: fixed

File: lib/server/routes/buckets.js [hunk 2]
```diff
@@ -1396,6 +1399,111 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+    return next(new errors.BadRequestError('Missing uploads'));
+  }
+
+  if (!Array.isArray(uploads)) {
+    return next(new errors.BadRequestError('Uploads is not an array'));
+  }
+
+  uploads.forEach(({ index, size }) => {
+    if (typeof size !== 'number' || size < 0) {
+      return next(new errors.BadRequestError('Invalid size'));
```
Comment (sg-gs) [inline lib/server/routes/buckets.js:1433]: Same as #52, be careful as this return will not end the execution nor the `foreach` neither `startUpload` function

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: lib/server/routes/buckets.js (hunk 2 above)
Comment (fabioespinosa) [reply lib/server/routes/buckets.js:1433]: Missed this completely, fixed

**Review by sg-gs — APPROVED** [review]

**Pull request merged**